test/MUXControlsTestApp/InfoBar.cs

```diff
@@ -0,0 +1,473 @@
+        }
+
+        public static readonly DependencyProperty CloseButtonCommandParameterProperty =
+            DependencyProperty.Register(nameof(CloseButtonCommandParameter), typeof(object), typeof(InfoBar), new PropertyMetadata(null));
+
+
+
+
+        /* Severity-Related Properties
+        * 
+        */
+        public InfoBarSeverity Severity
+        {
+            get { return (InfoBarSeverity)GetValue(SeverityProperty); }
+            set { SetValue(SeverityProperty, value); }
+        }
+
+        public static readonly DependencyProperty SeverityProperty =
+            DependencyProperty.Register(nameof(Severity), typeof(InfoBarSeverity), typeof(InfoBar), new PropertyMetadata(InfoBarSeverity.Default, OnPropertyChanged));
+
+        public Color StatusColor
+        {
+            get { return (Color)GetValue(StatusColorProperty); }
+            set { SetValue(StatusColorProperty, value); }
+        }
+
+        public static readonly DependencyProperty StatusColorProperty =
+            DependencyProperty.Register(nameof(StatusColor), typeof(Color), typeof(InfoBar), new PropertyMetadata(Color.FromArgb(0, 0, 0, 0)));
+
+
+        public IconSource IconSource
+        {
+            get { return (IconSource)GetValue(IconSourceProperty); }
+            set { SetValue(IconSourceProperty, value); }
+        }
+
+        public static readonly DependencyProperty IconSourceProperty =
+            DependencyProperty.Register(nameof(IconSource), typeof(IconSource), typeof(InfoBar), new PropertyMetadata(default));
+
+
+
+        // Event handlers for Close Button and Action Button
+        private void OnCloseButtonClick(object sender, RoutedEventArgs e)
+        {
+
+            lastCloseReason = InfoBarCloseReason.CloseButton;
+            InfoBarEventArgs newE = new InfoBarEventArgs();
+            if (CloseButtonClick != null)
+            {
+                CloseButtonClick.Invoke(this, newE);
+            }
+
+            //If the user sets IsHandled to true, they can override the behavior of CloseButtonClick
+            if (newE.IsHandled == false)
+            {
+                RaiseClosingEvent();
+                Open(IsOpen);
+                RaiseClosedEvent();
+            }
+
+        }
+
+
+        void RaiseClosingEvent()
+        {
+            InfoBarClosingEventArgs args = new InfoBarClosingEventArgs();
+            args.Reason = lastCloseReason;
+
+            if (Closing != null)
+            {
+                Closing.Invoke(this, args);
+            }
+
+            if (!args.Cancel)
+            {
+                _myBorder.Visibility = Visibility.Collapsed;
+                IsOpen = false;
+            }
+            else
+            {
+                // The developer has changed the Cancel property to true, indicating that they wish to Cancel the
+                // closing of this tip, so we need to revert the IsOpen property to true.
+                IsOpen = true;
```

Comment:
This shouldn't be necessary, since we have not done anything to set the IsOpen property to false yet, or am I missing something?